src/Spec-MorphicAdapters/UIThemeDecorator.class.st

```diff
@@ -0,0 +1,83 @@
+"
+I decorate a UITheme allowing to override desired properties.
+
+See #example on class side.
+"
+Class {
+	#name : #UIThemeDecorator,
+	#superclass : #Object,
+	#instVars : [
+		'theme',
+		'themeOverrideDict'
+	],
+	#category : #'Spec-MorphicAdapters-Support'
+}
+
+{ #category : #examples }
+UIThemeDecorator class >> exampleDecoratorToMakePluggableButtonRed [
+	| normalColorBlock clickedColorBlock |
+	normalColorBlock := [ :aButton |
+							(aButton valueOfProperty: #noFill ifAbsent: [false]) 
+								ifTrue: [ SolidFillStyle color: Color transparent ]
+								ifFalse: [ SolidFillStyle color: Color red ] ].
+	clickedColorBlock := [ :aButton | SolidFillStyle color: Color red muchDarker ].
+	^ (UIThemeDecorator theme: Smalltalk ui theme)
+			property: #buttonNormalFillStyleFor: returnsValueOf: normalColorBlock;
+			property: #buttonMouseOverFillStyleFor: returnsValueOf: normalColorBlock;
+			property: #buttonPressedFillStyleFor: returnsValueOf: clickedColorBlock;
+			yourself
+]
+
+{ #category : #'instance creation' }
+UIThemeDecorator class >> theme: aUITheme [
+	^ self new
+		theme: aUITheme;
+		yourself
+]
+
+{ #category : #'reflective operations' }
+UIThemeDecorator >> doesNotUnderstand: aMessage [
```

Comment:
Do we really need to override doesNotUnderstand:?

Reply:
Well, the goal is to let the UITheme decorated continue to answer to style messages that are not overridden by the decorator.
If you have another idea on how to do it without modifying the whole theme mechanism, I'd be happy to implement another solution.

Reply:
What I'm thinking about is that UIThemeDecorator should maybe inherit from ProtoObject since it is some kind of proxy after all.